app/api/routers/breeze_buddy/blacklist/handlers.py

```
@@ -0,0 +1,133 @@
+            merchant_id=request.merchant_id,
+            reason=request.reason,
+            created_by=current_user.username,
+        )
+
+        if result:
+            return result
+
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Failed to add number to blacklist. It may already be blacklisted.",
+        )
```

Ruff (0.15.2)

[warning] Abstract `raise` to an inner function

(TRY301)